test/e2e/mv3/service-worker-restart.spec.js

```diff
@@ -0,0 +1,184 @@
+        });
+
+        windowHandles = await driver.getAllWindowHandles();
+        const extension = windowHandles[0];
+        await driver.switchToWindow(extension);
+
+        // balance renders
+        await driver.waitForSelector(
+          {
+            css: '[class="eth-overview__primary-container"]',
+            // balance is 0 because we switched to an empty account
+            text: '0 ETH',
+          },
+          { timeout: 50_000 },
+        );
+
+        await driver.findElement({ text: '0x097...7950', tag: 'div' });
+
+        // assert that the segment request has been sent through inspecting the mock
+        await driver.wait(async () => {
+          const isPending = await mockedSegmentEndpoints.isPending();
+          return isPending === false;
+        }, 10_000);
+        const mockedRequests = await mockedSegmentEndpoints.getSeenRequests();
+
+        assert.equal(mockedRequests.length, numberOfSegmentRequests);
+
+        assert.equal(mockedRequests[0].url, 'https://api.segment.io/v1/batch');
+
+        assert.equal(mockedRequests[0].body.json.batch.length, 1);
+        assert.equal(
+          mockedRequests[0].body.json.batch[0].event,
+          EVENT_NAMES.SERVICE_WORKER_RESTARTED,
+        );
+
+        assert.equal(
+          mockedRequests[0].body.json.batch[0].properties.service_worker_action_queue_methods.indexOf(
+            'addNewAccount',
+          ) !== '-1',
+          true,
+        );
+        assert.equal(
+          mockedRequests[0].body.json.batch[0].properties.category,
+          EVENT.SOURCE.SERVICE_WORKERS,
+        );
+        assert.equal(
+          mockedRequests[0].body.json.batch[0].properties.category,
+          EVENT.SOURCE.SERVICE_WORKERS,
+        );
+        assert.equal(
+          mockedRequests[0].body.json.batch[0].properties.chain_id,
+          convertToHexValue(1337),
+        );
+        assert.equal(
+          mockedRequests[0].body.json.batch[0].properties.environment_type,
+          'background',
+        );
+        assert.equal(
+          mockedRequests[0].body.json.batch[0].properties.locale,
+          'en',
+        );
+
+        assert.equal(mockedRequests[1].url, 'https://api.segment.io/v1/batch');
+
+        assert.equal(mockedRequests[1].body.json.batch.length, 1);
+        assert.equal(
+          mockedRequests[1].body.json.batch[1].event,
+          EVENT_NAMES.SERVICE_WORKER_RESTARTED,
+        );
+
+        const serviceWorkerRestartTimeRequestProperties =
+          mockedRequests[1].body.json.batch[1].properties;
+        assert.equal(
+          serviceWorkerRestartTimeRequestProperties.category,
+          EVENT.SOURCE.SERVICE_WORKERS,
+        );
+        assert(
+          typeof serviceWorkerRestartTimeRequestProperties.service_worker_action_queue_methods ===
+            'number',
+        );
+        assert(
+          serviceWorkerRestartTimeRequestProperties.service_worker_action_queue_methods >
+            0,
+        );
```

Comment:
I think I missed something. I thought `service_worker_action_queue_methods` was an array. how is it a number here and an array in other places?